api/controller/triggers.go

```diff
@@ -14,6 +15,8 @@ import (
 
 const pageSizeUnlimited int64 = -1
 
+var idValidationPattern = regexp.MustCompile(`^[a-zA-Z0-9\-_]+$`)
```

Comment:
It is just question without suggestion: why you declared `idValidationPattern` at package level scope and not in `CreateTrigger` func? There is only one usage of var

Reply:
Because regex compilation is a heavy-weight operation. Our controller functions don't have any state, so I had to extract regex to global variable

Reply:
Nice, thank you!